src/chainparams.h

```diff
@@ -71,14 +72,62 @@ class CChainParams
     const std::vector<unsigned char>& Base58Prefix(Base58Type type) const { return base58Prefixes[type]; }
     const std::vector<SeedSpec6>& FixedSeeds() const { return vFixedSeeds; }
     const Checkpoints::CCheckpointData& Checkpoints() const { return checkpointData; }
+
+    /** Maximum block size of a block with timestamp nBlockTimestamp */
+    int ActivateSizeForkMajority() const { return nActivateSizeForkMajority; }
+    uint64_t SizeForkGracePeriod() const { return nSizeForkGracePeriod; }
+    uint64_t MaxBlockSize(uint64_t nBlockTimestamp) const {
```

Comment:
Could you please pass nSizeForkActivationTime in here, rather than changing the const assumption of Params()? main can be responsible for keeping tabs of the activation time.

It will make life much easier for libbitcoinconsensus. I should think that it would make it easier for testing as well, since you could more easily fake a time for a single test.